benchmark.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+const path = require('path');
+const Benchmark = require('benchmark');
+const makeDir = require('make-dir');
+const tempy = require('tempy');
+const del = require('.');
+
+const suite = new Benchmark.Suite('concurrency');
+
+const tempDir = tempy.directory();
+
+const fixtures = Array.from({length: 2000}, (x, index) => {
+	return path.resolve(tempDir, (index + 1).toString());
+});
+
+function createFixtures() {
+	for (const fixture of fixtures) {
+		makeDir.sync(path.resolve(tempDir, fixture));
+	}
+}
+
+const concurrency = [1, 3, 5, 10, 15, 20, 50, 100, Infinity];
+
+concurrency.forEach(num => {
+	const name = `concurrency: ${num.toString()}`;
+
+	suite.add({
+		name,
+		defer: true,
+		fn(deferred) {
+			// Can't use setup because it isn't called after every defer
+			// https://github.com/bestiejs/benchmark.js/issues/136
+			createFixtures();
+
+			// Async await was giving too many errors. stick with standard promises
```

Comment:
Huh? What does this mean? Async/await should work any differently.

Reply:
```js
suite.add({
	name,
	defer: true,
    // if you remove setup() the benchmark works as expected
	setup() {
	},
	async fn(deferred) {
		// Can't use setup because it isn't called after every defer
		// https://github.com/bestiejs/benchmark.js/issues/136
		createFixtures();

		// Async await was giving too many errors. stick with standard promises
		await del(['**/*'], {
			cwd: tempDir,
			concurrency,
			// eslint-disable-next-line promise/prefer-await-to-then
		}).then(removedFiles => {
			if (removedFiles.length !== fixtures.length) {
				const error = new Error(
					`"${name}": files removed: ${removedFiles.length}, expected: ${fixtures.length}`,
				);

				console.error(error);

				del.sync(tempDir, { cwd: tempDir, force: true });

				// eslint-disable-next-line unicorn/no-process-exit
				process.exit(1);
			}

			deferred.resolve();
		});
	},
});
```

Gives the error:
```js
del on  add-benchmarks [⇡!] is 📦 v5.0.0 via ⬢ v10.16.0 took 6s
➜ node benchmark.js
undefined:9
			await del(['**/*'], {
			^^^^^

SyntaxError: await is only valid in async function
    at Function (<anonymous>)
    at createCompiled (/Users/chris/github/del/node_modules/benchmark/benchmark.js:1725:16)
    at clock (/Users/chris/github/del/node_modules/benchmark/benchmark.js:1608:58)
    at clock (/Users/chris/github/del/node_modules/benchmark/benchmark.js:1818:20)
    at new Deferred (/Users/chris/github/del/node_modules/benchmark/benchmark.js:405:7)
    at Deferred (/Users/chris/github/del/node_modules/benchmark/benchmark.js:402:16)
    at Benchmark.run (/Users/chris/github/del/node_modules/benchmark/benchmark.js:2112:13)
    at execute (/Users/chris/github/del/node_modules/benchmark/benchmark.js:860:74)
    at Timeout._onTimeout (/Users/chris/github/del/node_modules/lodash/lodash.js:2750:43)
    at ontimeout (timers.js:436:11)
```

Reply:
Weird. I already commented on this, but it's gone now... Anyway, I think you're just missing the `async` keyword for the method. Then it worked for me at least.